modules/stochastic_tools/include/reporters/ActiveLearningReporterBase.h

```diff
@@ -0,0 +1,152 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "StochasticReporter.h"
+
+template <typename T>
+class ActiveLearningReporterTempl;
+
+/**
+ * This is a base class for performing active learning routines, meant to be used
+ * in conjunction with Sampler multiapps and SamplerReporterTransfer. The purpose
+ * is to determine if a sample needs to be evaluated by a multiphysics solve and
+ * possibly replace quantities of interest with values computed by other means.
+ */
+template <typename T>
```

Comment:
Is the class templated only because _default_value is templated?   Is default_value ever anything other than a scalar?

Reply:
No, it is templated based on the type of reporter being transferred. We may want in the future to have a QoI be a vector, in such a case you will want to derive from the `ActiveLearningReporterTempl<std::vector<Real>>` version of this class.